wgengine/netstack/netstack.go

```diff
@@ -242,14 +242,28 @@ func (ns *Impl) acceptTCP(r *tcp.ForwarderRequest) {
 
 func (ns *Impl) forwardTCP(client *gonet.TCPConn, address string) {
 	ns.logf("[v2] netstack: forwarding to address %s", address)
-	server, err := ns.dialTCP(address)
+	ctx, cancel := context.WithCancel(context.Background())
+	server, err := ns.dialContextTCP(ctx, address)
 	if err != nil {
 		ns.logf("netstack: could not connect to server %s: %s", address, err)
 		client.Close()
+		cancel()
```

Comment:
why don't you make this a `defer cancel()` instead after line 245, then remove this line and line 264.

likewise with server.Close and client.Close on 265-266